lib/chargify_api_ares/resources/invoice.rb

```diff
@@ -9,27 +9,34 @@ class Payment < Base
     end
 
     def self.find_by_invoice_id(id)
-      find(:first, {:params => {:id => id}})
+      find(:first, { params:  { id: } })
     end
 
-    def self.find_by_subscription_id(id)
-      find(:all, {:params => {:subscription_id => id}})
+    def self.find_by_subscription_id(subscription_id)
+      find(:all, { params:  { subscription_id: } })
+    end
+
+    def self.find_by_subscription_and_date_period(subscription_id, start_date, end_date, extra_params = {})
```

Comment:
it's weird calling it `find_by` if it will return a list

Reply:
I know, I followed the same pattern than `find_by_subscription_id`. I can change it by `list_by_subscription_and_date_period` or something like that

Reply:
Makes sense